public/js/chat-cmds.js

```diff
@@ -213,3 +209,103 @@ var DashboardToggle = function() {
     return this;
 };
 
+var ChatCommands = {
+    _cmds: {},
+
+    _chatObject: null,
+
+    /**
+     * Initializes the chat commands object.
+     *
+     * @param {Object} chatObj - Chat object.
+     * @returns {undefined} undefined
+     * @public
+     */
+    initialize: function(chatObj) {
+        this._chatObject = chatObj;
+
+        // @todo Add new plugins here
+        this.registerCommand(new ListUser());
+        this.registerCommand(new CameraToggle());
+        this.registerCommand(new MicToggle());
+        this.registerCommand(new DashboardToggle());
+    },
+
+    /**
+     * Registers a chat command.
+     *
+     * @param {Object} obj - Chat command object.
+     * @returns {undefined} undefined
+     * @public
+     */
+    registerCommand: function(obj) {
+        var cmd = obj.command.toLowerCase();
+        this._cmds[cmd] = obj;
+    },
+
+    /**
+     * Handles a chat command. Must return true or false to
+     * indicate whether the command was handled or not.
+     *
+     * @param {String} message - Chat command message.
+     * @returns {Boolean} True if the command was handled,
+     * false otherwise.
+     * @public
+     */
+    handleCommand: function(message) {
+        if (!message.length) {
+            ErrorMetric.log('ChatCommands.handleCommand => message too small');
+            ErrorMetric.log('                              msg: "' + message + '"');
+            return false;
+        }
+
+        var argv = ShellQuote.parse(message.substring(1));
+        if (!argv.length) {
+            ErrorMetric.log('ChatCommands.handleCommand => bad message');
+            ErrorMetric.log('                              msg: "' + message + '"');
+            return false;
+        }
+
+        var cmd = argv[0].toLowerCase();
+        if (cmd === 'help') {
+            var content = '<div class="chatInternal">';
+            content += '<h1>Chat Commands</h1>\n<ul>';
+            content += '<li><h2><span class="chatIntCmdName">/help</span></h2>' +
+                       '<p>Shows this message</p></li>';
+            for (var i in this._cmds) {
+                if (this._cmds.hasOwnProperty(i)) {
```

Comment:
Included a `hasOwnProperty` check in order to filter out properties on the prototype chain